nixos/modules/services/misc/jellyfin.nix

```diff
@@ -35,7 +35,26 @@ in
         Group = cfg.group;
         StateDirectory = "jellyfin";
         CacheDirectory = "jellyfin";
-        ExecStart = "${pkgs.jellyfin}/bin/jellyfin --datadir '/var/lib/${StateDirectory}' --cachedir '/var/cache/${CacheDirectory}'";
+        db_version = pkgs.lib.versions.majorMinor pkgs.jellyfin.version;
+        ExecPreStart = ''
+          #!/bin/bash
+          mkdir -p /var/lib/${StateDirectory}/database/${db_version}
+          if [ "${db_version}" = "10.6.1" ]; then
+            cp -r /var/lib/${StateDirectory}/data /var/lib/${StateDirectory}/database/${db_version}/
```

Comment:
I have not checked, but what happens to everything that does not fall into `data`? (`log` and so on)